lib/ex_machina/ecto_strategy.ex

```diff
@@ -74,18 +74,17 @@ defmodule ExMachina.EctoStrategy do
     assocs = get_schema_assocs(schema)
 
     Enum.reduce(assocs, struct, fn(assoc, struct) ->
-      original_value = Map.get(struct, assoc)
-      casted_value = if is_list(original_value) do
-        Enum.map(original_value, &(cast_assoc(&1, assoc, struct)))
-      else
-        cast_assoc(original_value, assoc, struct)
-      end
+      casted_value = struct |> Map.get(assoc) |> cast_assoc(assoc, struct)
+
       Map.put(struct, assoc, casted_value)
     end)
   end
 
-  defp cast_assoc(original_value, assoc, %{__struct__: schema}) do
+  defp cast_assoc(original_value, assoc, %{__struct__: schema} = struct) do
     case original_value do
+      original_value when is_list(original_value) ->
```

Comment:
Hmmm. yeah. It looks a bit weird. Fixed in e03690a6bd357c11dc0b0d5c2df648686f6a96a1